### PR TITLE
Notes about DI usage, and constant reassignment

### PR DESCRIPTION
**Pull request by kzaitsev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ A note on the language:
 * Use the `.ruby-version` file convention to specify the Ruby version for a project.
 * Use maximum 120 characters in a single line.
 * Use `Class.new(StandardError)` instead of inheritance to declare a single-line error class.
+* Use Dependency Injection pattern instead of specifying a hardcoded behavior for the environment.
```
Comment (exAspArk) [inline README.md]: I'd say `Prefer` :)
Generally using specific patterns should not be mandatory

Comment (kzaitsev) [reply README.md]: done

**Comment by exAspArk** [comment]
Need to add some ⚓️  :)

**Comment by linchus** [comment]
👍

**Pull request merged**